[JDK20] Fat Failure: com.ibm.ws.logging.hpel_fat: testbinaryLogCopyExecutes

RTC 294161

```
junit.framework.AssertionFailedError: 2022-12-12-17:49:31:442 Failed assertion that binaryLog exited with successful return code
at com.ibm.ws.logging.hpel.fat.BinaryLogExec.testbinaryLogCopyExecutes(BinaryLogExec.java:203)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:201)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:354)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:175)
```

```
[01/19/2023 01:45:54:204 PST] 001 FATRunner                      evaluate                       I entering com.ibm.ws.logging.hpel.fat.BinaryLogExec.testbinaryLogCopyExecutes
[01/19/2023 01:45:54:204 PST] 001 FATRunner                      retrieveFFDCCounts             I FFDC log collection for server: HpelServer : componenttest.topology.impl.LibertyServer@149f5761 is NOT skipped. FFDC Checking is enabled for this server.
[01/19/2023 01:45:55:100 PST] 001 BinaryLogExec                  logMsg                         I     === Current environment's locale setting : en_US
[01/19/2023 01:45:55:104 PST] 001 DirectMethodHandleAccessor     invoke                         I executing binaryLog on HpelServer
[01/19/2023 01:45:55:106 PST] 001 LibertyServer                  getFileFromLiberty             I Getting file: /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/HpelServer/logs
[01/19/2023 01:45:55:106 PST] 001 BinaryLogExec                  logMsg                         I executing: /home/jazz_build/Build/jbe/build/dev/image/output/wlp/bin/binaryLog "copy" "/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/HpelServer/logs" "/home/jazz_build/Build/jbe/build/dev/build.image/build/tmp/com.ibm.ws.logging.hpel.fat.BinaryLogExec_1674121554203.lv_out.log" "--minDate=01/19/23, 01:45:54:205 PST:205" "--maxDate=2023-01-19T01:45:55.100-0800" 
[01/19/2023 01:45:55:948 PST] 001 DirectMethodHandleAccessor     invoke                         I Verifying binaryLog std out/err and status return code.
[01/19/2023 01:45:55:948 PST] 001 BinaryLogExec                  logMsg                         I     === BinaryLog's stdout: ===  return code is : 20
[01/19/2023 01:45:55:949 PST] 001 BinaryLogExec                  logMsg                         I Unable to parse --minDate value.
For usage information use binaryLog help.

[01/19/2023 01:45:55:949 PST] 001 BinaryLogExec                  logMsg                         I  
[01/19/2023 01:45:55:949 PST] 001 FATRunner                      evaluate                       S null
java.lang.AssertionError: Failed assertion that binaryLog exited with successful return code
	at org.junit.Assert.fail(Assert.java:93)
	at org.junit.Assert.assertTrue(Assert.java:43)
	at com.ibm.ws.logging.hpel.fat.BinaryLogExec.testbinaryLogCopyExecutes(BinaryLogExec.java:205)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:203)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:30)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:356)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:177)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:24)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:39)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:520)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1060)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:911)
[01/19/2023 01:45:55:951 PST] 001 FATRunner                      evaluate                       I exiting com.ibm.ws.logging.hpel.fat.BinaryLogExec.testbinaryLogCopyExecutes
[01
```

--- a/dev/com.ibm.ws.logging.hpel/src/com/ibm/ws/logging/hpel/FormatSet.java
+++ b/dev/com.ibm.ws.logging.hpel/src/com/ibm/ws/logging/hpel/FormatSet.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2009, 2017 IBM Corporation and others.
+ * Copyright (c) 2009, 2023 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -147,6 +147,7 @@ public class FormatSet {
                 newPattern = newPattern.replace('K', 'H');
                 newPattern = newPattern.replace('k', 'H');
                 newPattern = newPattern.replace('a', ' ');
+                newPattern = newPattern.replace('\u202F', ' ');
                 newPattern = newPattern.trim();
                 sdFormatter.applyPattern(newPattern);
                 formatter = sdFormatter;

